src/main/java/com/google/devtools/build/lib/analysis/platform/PlatformUtils.java

```diff
@@ -67,6 +67,29 @@ public static Platform getPlatformProto(Spawn spawn, @Nullable RemoteOptions rem
     return getPlatformProto(spawn, remoteOptions, ImmutableMap.of());
   }
 
+  private static boolean shouldProducePlatformProto(
+      Spawn spawn, SortedMap<String, String> defaultExecProperties, Map<String, String> additionalProperties) {
+    PlatformInfo executionPlatform = spawn.getExecutionPlatform();
+    if (executionPlatform != null) {
+      if (!executionPlatform.execProperties().isEmpty()) {
```

Comment:
These conditions look right to me but just to clarify: are they part of your proposal or just an incidental cleanup?

Reply:
This change was prompted by https://github.com/bazelbuild/bazel/pull/24265#discussion_r1949436845 ([context](https://github.com/bazelbuild/bazel/commit/88395ce76f1904411bb9e7228c44681dcf69637a)).

From memory, `spawn.getExecutionPlatform().remoteExecutionProperties().isEmpty()` was added as removal of the null platform in `src/test/java/com/google/devtools/build/lib/exec/util/FakeOwner.java` resulted in several tests breaking due to empty platform protos appearing as a result.